### DISCUSSION TITLE
Ring smart lighting bridge and transformer not showing up

### DISCUSSION THREAD

**Discussion by mrhdsz** [tsightler/ring-mqtt#831] [Support Q&A] [answered]
I added a Ring smart lighting bridge (https://ring.com/products/smart-lighting-bridge) and a Ring low-voltage transformer that connects through the bridge. After getting those set up correctly in the Ring app, they show up as devices in the device list, and all controls work properly in the Ring app as well. However, these don't show up at all in through ring-mqtt in Home Assistant. I have a default install of both mosquitto broker and the ring-mqtt add-on. Has anyone else gotten these types of devices working through MQTT? 

I have turned on debugging and looked through logs, as well as after a fresh reboot of HA, but logs have no errors or events that show up having to do with the smart lighting or bridge.

**Comment by tsightler** [comment]
Please make sure you restart the addon, not HA, restarting HA does nothing.

**Comment by mrhdsz** [comment]
I did try that now, and still no luck. I switched over to EMQX instead of Mosquitto so I could have some more visibility into the MQTT topics. Looks like there is one coming through for "ring/{id}/lighting/{id}/light/command.

**Reply by tsightler** [reply]
In my experience changing default setups rarely lead to success.  You can simply use tools like MQTT Explorer (either standalone or via HA addon) to see MQTT messages.  However, logs in the addon show everything that is happening, including discovered devices so rarely is it needed to watch messages externally.

**Reply by mrhdsz** [reply]
Thank you for the quick help on this by the way, I really appreciate it! I do see in the logs from ring-mqtt that the bridge ("Big Woods Bridge") and lights ("Side Yard Lights") are being picked up, but for some reason they don't show as a device or entity in Home Assistant. Below is a snippet from the logs (I removed all the IDs, not sure if those are generated or actually tied to my devices so didn't want to post them).  

`2024-05-08T18:44:51.472Z ring-attr [Big Woods Bridge] ring/[id]/alarm/[id]/info/state {"commStatus":"ok","lastCommTime":"1970-01-01T00:00:00Z","lastUpdate":"2024-05-08T18:02:03Z","tamperStatus":"ok","wirelessSignal":-49}
2024-05-08T18:44:51.472Z ring-attr [Big Woods Bridge] ring/[id]/alarm/[id]/wireless/attributes {"wirelessSignal":-49}
2024-05-08T18:44:51.472Z ring-attr [Side Yard Lights] ring/[id]/lighting/[id]/info/state {"acStatus":"ok","commStatus":"ok","firmwareStatus":"up-to-date","lastCommTime":"2024-05-08T18:43:54Z","lastUpdate":"2024-05-08T18:43:54Z","serialNumber":"[removed]","tamperStatus":"ok"}`

**Reply by tsightler** [reply, marked as answer]
Those are just state updates, you need to look for discovery messages which are sent to the homeassistant discovery topic.  These will be sent shortly after the discovered devices are listed in the ring-mqtt logs.  Also, in the HA MQTT integration you must have discovery enabled and the default discovery topic (homeassistant).  You should be able to see the discovery messages being sent in the addon logs, and the devices being received in the HA logs (you may need to enable additional debugging for HA).

That's all there is to device discovery.  Honestly, about 90% of the discovery issues I've seen are with browser refresh and have nothing to do with discovery at all, so be sure to force a hard refresh in your browser (typically shift-f5) as sometimes devices will not appear due to caching of device list browser side.

**Reply by mrhdsz** [reply]
Got it working. I completely removed mosquitto and reinstalled it, set it up with a new user, and then reconfigured ring-mqtt with that user. After restarting everything again, the discovery process worked and picked up the bridge and lights. Thanks for the help!
